pipeline-adapters/mlrun-pipelines-kfp-v1-8/src/mlrun_pipelines/models.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+from typing import Any, Union
+
+from kfp.dsl import ContainerOp
+from kfp_server_api.models.api_run_detail import ApiRunDetail
+from mlrun_pipelines.common.helpers import FlexibleMapper
+
+# class pointer for type checking on the main MLRun codebase
+PipelineNodeWrapper = ContainerOp
+
+
+class PipelineManifest(FlexibleMapper):
+    def __init__(
+        self, workflow_manifest: Union[str, dict] = "{}", pipeline_manifest: str = "{}"
+    ):
+        try:
+            main_manifest = json.loads(workflow_manifest)
+        except TypeError:
+            main_manifest = workflow_manifest
+        if pipeline_manifest:
+            pipeline_manifest = json.loads(pipeline_manifest)
+            main_manifest["status"] = pipeline_manifest.get("status", {})
+        super().__init__(main_manifest)
+
+
+class PipelineRun(FlexibleMapper):
```

Comment:
Is this class planned to be different in KFP 2?
If not, it might reside in the common package

Reply:
Yeah, the implementation code will be very different, but the class name, methods, and arguments will be identical. So MLRun will call the v2 version the same way it does with this one. 

Neat stuff if I do say so myself 😅